lib/index.js

```diff
@@ -263,16 +312,76 @@ api.get = async ({actor, account, email, type} = {}) => {
   } else {
     query['account.email'] = email;
   }
+
   const record = await database.collections.account.findOne(
     query, {projection: {_id: 0, [typeKey]: 1}});
+
   if(!record) {
     throw new BedrockError('Authentication token not found.', 'NotFoundError', {
       httpStatusCode: 404,
       public: true
     });
   }
 
-  return record.meta['bedrock-authn-token'].tokens[type];
+  const result = record.meta['bedrock-authn-token'].tokens[type];
+  if(type === 'nonce') {
+    if(tokenId) {
+      return result.filter(token => token.id === tokenId)[0];
```

Comment:
Can we put the `id` in the query, if it is present, such that we'll hit the "Authentication token not found" above and we don't have to repeat it down here? We'll still need to filter to find the token to return, but we'll know it's there if we hit this path.

I also think it should apply to all token types as well, not just nonces. That should further simplify.